ada_feeding/ada_feeding/behaviors/acquisition/compute_food_frame.py

```
@@ -5,56 +5,246 @@
+                                                        static broadcaster here to keep it alive
+                                                        for debugging purposes.
+        """
+        # pylint: disable=unused-argument
+        # Arguments are handled generically in base class.
+        super().blackboard_outputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    # pylint: disable=attribute-defined-outside-init
+    # It is okay for attributes in behaviors to be
+    # defined in the setup / initialise functions.
+
+    def setup(self, **kwargs):
+        """
+        Middleware (i.e. TF) setup
+        """
+
+        # Create TF Tree
+        self.tf_buffer = Buffer()
+        self.tf_listener = TransformListener(
+            self.tf_buffer, self.blackboard_get("ros2_node")
+        )
 
-    def setup(self, **kwargs) -> None:
-        pass
+    def initialise(self):
+        """
+        Behavior initialization
+        """
+
+        # Construct camera intrinsics
+        # See: https://medium.com/@yasuhirachiba/converting-2d-image-coordinates-
+        #      to-3d-coordinates-using-ros-intel-realsense-d435-kinect-88621e8e733a
+        camera_info = self.blackboard_get("camera_info")
+        self.intrinsics = pyrealsense2.intrinsics()
+        self.intrinsics.width = camera_info.width
+        self.intrinsics.height = camera_info.height
+        self.intrinsics.ppx = camera_info.k[2]
+        self.intrinsics.ppy = camera_info.k[5]
+        self.intrinsics.fx = camera_info.k[0]
+        self.intrinsics.fy = camera_info.k[4]
+        if camera_info.distortion_model == "plumb_bob":
+            self.intrinsics.model = pyrealsense2.distortion.brown_conrady
+        elif camera_info.distortion_model == "equidistant":
+            self.intrinsics.model = pyrealsense2.distortion.kannala_brandt4
+        else:
+            self.intrinsics.model = pyrealsense2.distortion.none
```

Should we log an error here? This is unlikely to arise for our RealSense, which is "plumb_bob", but I'm wondering if we should raise an error in case others reuse our code.

---

I added a warning.